src/sentry/integrations/github/webhook.py

```diff
@@ -259,18 +248,16 @@ def _handle(
                                 # even if we can't find a user, set to none so we
                                 # don't re-query
                                 gh_username_cache[gh_username] = None
-                                try:
-                                    identity = Identity.objects.get(
-                                        external_id=gh_user["id"],
-                                        idp__type=self.provider,
-                                        idp__external_id=self.get_idp_external_id(
-                                            integration, host
-                                        ),
-                                    )
-                                except Identity.DoesNotExist:
-                                    pass
-                                else:
-                                    author_email = identity.user.email
+                                identity_user = None
+                                identity = identity_service.get_identity(
+                                    identity_ext_id=gh_user["id"],
+                                    provider_type=self.provider,
+                                    provider_ext_id=self.get_idp_external_id(integration, host),
+                                )
+                                if identity is not None:
+                                    identity_user = user_service.get_user(user_id=identity.user_id)
```

Comment:
At some point in the future, we might want to have a way to make this a single RPC call. Perhaps the `identity_service` could inline the connected user, as the Identity by itself isn't that helpful.

Reply:
I'll add a TODO here as a note.